herddb-core/src/main/java/herddb/core/TableManager.java

```diff
@@ -1584,18 +1585,26 @@ private void applyDelete(Bytes key) throws DataStorageManagerException {
         if (indexes == null) {
             /* We don't need the page if isn't loaded or isn't a mutable new page */
             page = newPages.get(pageId);
-            previous = null;
 
             if (page != null) {
                 pageReplacementPolicy.pageHit(page);
+                previous = page.get(key);
```

Comment:
this is a behavior change. this is not only logs

Reply:
Yes I know. I cant split the commit if needed. This code part should be equal to update one. It's about old record retrieval to update dirty status as precise as possible without load data if there isn't

Reply:
ok